### PR TITLE
Feature/faml703 implement req session

### PR DESCRIPTION
**Pull request by ICraig-CH**
Apologies for the large pr, the changes are as follows (and do follow the dummy pr) : 
- removed old session code
- added cache_service.js to get cached data and to save to the session
- updated CH node session handler version
- fixed broken imports
- fixed code that broke because of old implementation
- fixed tests and added new tests for cache_service.js

Jira ticket : https://companieshouse.atlassian.net/browse/FAML-703

### CODE REVIEW

File: server/routes/report.js [hunk 1]
```diff
@@ -22,11 +22,11 @@ let selfLink; // eslint-disable-line no-unused-vars
...
   try {
-    session = new Session(req, res);
-    if (typeof res.locals.session.appData.initialServiceResponse === 'undefined') {
+    if (typeof cacheService.getCachedDataFromSession(req.session).appData.initialServiceResponse === 'undefined') {
       selfLink = '';
     } else {
-      selfLink = res.locals.session.appData.initialServiceResponse.links.self;
+      const cacheData = cacheService.getCachedDataFromSession(req.session);
+      selfLink = cacheData.appData.initialServiceResponse.links.self;
     }
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:30]: Could inline this, although that might be a bit long.

File: server/routes/report.js [hunk 2]
```diff
@@ -332,15 +329,14 @@ router.post('/report-a-discrepancy/psc-name', (req, res) => {
...
+      const pscCache = cacheService.getCachedDataFromSession(req.session);
+      pscCache.appData.selectedPscDetails = pscDetails;
+      cacheService.setPscCache(req.session, pscCache);
     }).then(_ => {
       res.redirect(302, viewData.path);
     }).catch(err => {
-      viewData.this_data.organisationName = res.locals.session.appData.companyName;
-      viewData.title = routeUtils.setPageTitle('Which PSC has a discrepancy for ' + res.locals.session.appData.companyName + '?');
+      viewData.this_data.organisationName = cacheService.getCachedDataFromSession(req.session).appData.companyName;
+      viewData.title = routeUtils.setPageTitle('Which PSC has a discrepancy for ' + cacheService.getCachedDataFromSession(req.session).appData.companyName + '?');
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:339]: move the method call into a variable?

File: server/routes/utils/authentication.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 const Utility = require(`${serverRoot}/lib/Utility`);
-const { SessionKey } = require('ch-node-session-handler/lib/session/keys/SessionKey');
-const { SignInInfoKeys } = require('ch-node-session-handler/lib/session/keys/SignInInfoKeys');
+const { SessionKey } = require('@companieshouse/node-session-handler/lib/session/keys/SessionKey');
+const { SignInInfoKeys } = require('@companieshouse/node-session-handler/lib/session/keys/SignInInfoKeys');
 
 const authentication = (req, res, next) => {
   try {
-    const authCheck = (req.session && req.session.__value && req.session.__value.data && req.session.__value.data[SessionKey.SignInInfo])
-      ? req.session.__value.data[SessionKey.SignInInfo][SignInInfoKeys.SignedIn] === 1
```
Comment (CH-JonathanBlake) [inline server/routes/utils/authentication.js:8]: Yeah, get lost .__value's this makes me happy to see. :D

**Review by CH-JonathanBlake — APPROVED** [review]
Approved, but since I was involved in this it might not count for much.

File: server/routes/report.js (hunk 2 above)
Comment (ICraig-CH) [reply server/routes/report.js:339]: I can move that now

File: server/routes/report.js (hunk 1 above)
Comment (ICraig-CH) [reply server/routes/report.js:30]: to be fair it was like that before, so I can change that its not too much longer

File: server/routes/report.js [hunk 4]
```diff
@@ -268,6 +267,9 @@ router.get('/report-a-discrepancy/psc-name', (req, res) => {
     })
     .then(profile => {
       viewData.this_data.organisationName = profile.resource.companyName;
+      const pscCacheCompanyName = cacheService.getCachedDataFromSession(req.session);
+      pscCacheCompanyName.appData.companyName = profile.resource.companyName;
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:271]: Maybe a better name for this exists.

**Review by ch-rfishwick — APPROVED** [review]

**Review by Zain-Abbas1 — APPROVED** [review]

**Pull request merged**